src/main/java/com/tp/tanks/controller/UserController.java

```
@@ -6,61 +6,98 @@
             return new ResponseEntity<>(HttpStatus.FORBIDDEN);
         }
 
+        this.logger.info("[signUp] OUTPUT: username = " + user.getUsername() + " email = " + user.getEmail());
+
         session.setAttribute("userId", saveUser.getId());
         return new ResponseEntity<>(saveUser, HttpStatus.CREATED);
     }
 
+    @CrossOrigin
     @RequestMapping(value = "/signIn", method = RequestMethod.POST,
             consumes = "application/json", produces = "application/json")
     public ResponseEntity<User> signIn(@RequestBody User user, HttpSession session) {
 
+        this.logger.info("[signIn] INPUT:  username = " + user.getUsername() + " email = " + user.getEmail());
+
         final User loginUser = userService.signIn(user);
         if (loginUser == null) {
+            this.logger.error("[signIn] loginUser == null");
             return new ResponseEntity<>(HttpStatus.FORBIDDEN);
         }
 
+        this.logger.info("[signIn] OUTPUT: username = " + loginUser.getUsername() + " email = " + loginUser.getEmail());
+
         session.setAttribute("userId", loginUser.getId());
 
         return new ResponseEntity<>(loginUser, HttpStatus.OK);
     }
 
+    @CrossOrigin
     @RequestMapping(value = "/logout", method = RequestMethod.GET, produces = "application/json")
     public ResponseEntity logout(HttpSession session) {
 
-        session.removeAttribute("userId");
-        return new ResponseEntity<>(HttpStatus.OK);
+        if (session.getAttribute("userId") != null) {
+            this.logger.info("[logout]");
+            session.removeAttribute("userId");
+            return new ResponseEntity<>(HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(HttpStatus.FORBIDDEN);
     }
 
+    @CrossOrigin
     @RequestMapping(value = "/profile", method = RequestMethod.GET, produces = "application/json")
     public ResponseEntity getProfile(HttpSession session) {
 
         final Object id = session.getAttribute("userId");
```

Зачем брать Object почему нельзя преобразовать тут - сразу будет fail fast если что